maintenance/wikia/deleteImageRevision.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+require_once( __DIR__ . '/../Maintenance.php' );
+
+class DeleteImageRevision extends Maintenance {
+
+	const PAGE_ID_OPTION = 'pageId';
+	const REVISION_ID_OPTION = 'revisionId';
+	const REASON_OPTION = 'reason';
+
+	public function __construct() {
+		parent::__construct();
+		$this->addOption( self::PAGE_ID_OPTION, "image page id", true );
+		$this->addOption( self::REVISION_ID_OPTION, "revision id", true );
+		$this->addOption( self::REASON_OPTION, "deletion reason", true );
+		$this->mDescription = "Remove given revision of given image file";
+	}
+
+
+	public function execute() {
+		$pageId = intval( $this->getOption( self::PAGE_ID_OPTION ) );
+		$revisionId = intval( $this->getOption( self::REVISION_ID_OPTION ) );
+		$reason = $this->getOption( self::REASON_OPTION );
+		$title = Title::newFromID( $pageId );
+
+		//TODO: fix it, adding description to imagepage creates new revision so following check will not check if the image is old
+		if ( $revisionId < $title->getLatestRevID() ) {
+			$this->removeOldRevision( $title, $revisionId, $pageId, $reason );
+		} else {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS108)